File: app/src/main/java/com/jmalinen/blackjack/ui/components/GameInfoBar.kt
package com.jmalinen.blackjack.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.foundation.layout.asPaddingValues
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.statusBars
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.OutlinedButton
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.jmalinen.blackjack.ui.theme.FeltGreenDark
import com.jmalinen.blackjack.ui.theme.GoldAccent

@Composable
fun GameInfoBar(
    chips: Int,
    currentBet: Int,
    shoePenetration: Float,
    handsPlayed: Int,
    handsWon: Int,
    coachEnabled: Boolean,
    onToggleCoach: () -> Unit,
    deviationsEnabled: Boolean,
    onToggleDeviations: () -> Unit,
    showCount: Boolean,
    runningCount: Int,
    trueCount: Float,
    onToggleCount: () -> Unit,
    onShowChart: () -> Unit,
    onEnd: () -> Unit,
    modifier: Modifier = Modifier
) {
    val statusBarPadding = WindowInsets.statusBars.asPaddingValues()
    Column(
        modifier = modifier
            .fillMaxWidth()
            .background(FeltGreenDark)
            .padding(top = statusBarPadding.calculateTopPadding())
            .padding(horizontal = 16.dp, vertical = 8.dp)
    ) {
        Row(
            modifier = Modifier.fillMaxWidth(),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically
        ) {
            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(
                    text = "Chips:",
                    color = Color.White.copy(alpha = 0.7f),
                    fontSize = 14.sp
                )
                Spacer(modifier = Modifier.width(4.dp))
                Text(
                    text = "\$$chips",
                    color = GoldAccent,
                    fontSize = 18.sp,
                    fontWeight = FontWeight.Bold
                )
            }

            Text(
                text = "W: $handsWon / $handsPlayed",
                color = Color.White.copy(alpha = 0.7f),
                fontSize = 13.sp
            )

            Row(horizontalArrangement = Arrangement.spacedBy(6.dp)) {
                val coachColor = if (coachEnabled) Color(0xFF4CAF50) else Color.White.copy(alpha = 0.4f)
                Text(
                    text = "Coach",
                    color = coachColor,
                    fontSize = 12.sp,
                    fontWeight = if (coachEnabled) FontWeight.Bold else FontWeight.Normal,
                    modifier = Modifier
                        .clip(RoundedCornerShape(12.dp))
                        .then(
                            if (coachEnabled) Modifier.background(Color(0xFF4CAF50).copy(alpha = 0.2f))
                            else Modifier.border(1.dp, Color.White.copy(alpha = 0.3f), RoundedCornerShape(12.dp))
                        )
                        .clickable { onToggleCoach() }
                        .padding(horizontal = 8.dp, vertical = 3.dp)
                )

                if (coachEnabled) {
                    val devColor = if (deviationsEnabled) Color(0xFFFF9800) else Color.White.copy(alpha = 0.4f)
                    Text(
                        text = "Dev",
                        color = devColor,
                        fontSize = 12.sp,
                        fontWeight = if (deviationsEnabled) FontWeight.Bold else FontWeight.Normal,
                        modifier = Modifier
                            .clip(RoundedCornerShape(12.dp))
                            .then(
                                if (deviationsEnabled) Modifier.background(Color(0xFFFF9800).copy(alpha = 0.2f))
                                else Modifier.border(1.dp, Color.White.copy(alpha = 0.3f), RoundedCornerShape(12.dp))
                            )
                            .clickable { onToggleDeviations() }
                            .padding(horizontal = 8.dp, vertical = 3.dp)
                    )
                }

                val countColor = if (showCount) Color(0xFF42A5F5) else Color.White.copy(alpha = 0.4f)
                val trueCountStr = if (trueCount >= 0) "+%.1f".format(trueCount) else "%.1f".format(trueCount)
                val rcStr = if (runningCount >= 0) "+$runningCount" else "$runningCount"
                Text(
                    text = if (showCount) "RC:$rcStr TC:$trueCountStr" else "Count",
                    color = countColor,
                    fontSize = 12.sp,
                    fontWeight = if (showCount) FontWeight.Bold else FontWeight.Normal,
                    modifier = Modifier
                        .clip(RoundedCornerShape(12.dp))
                        .then(
                            if (showCount) Modifier.background(Color(0xFF42A5F5).copy(alpha = 0.2f))
                            else Modifier.border(1.dp, Color.White.copy(alpha = 0.3f), RoundedCornerShape(12.dp))
                        )
                        .clickable { onToggleCount() }
                        .padding(horizontal = 8.dp, vertical = 3.dp)
                )

            }

            Row(verticalAlignment = Alignment.CenterVertically) {
                Text(
                    text = "Bet:",
                    color = Color.White.copy(alpha = 0.7f),
                    fontSize = 14.sp
                )
                Spacer(modifier = Modifier.width(4.dp))
                Text(
                    text = "\$$currentBet",
                    color = Color.White,
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Bold
                )
            }
        }

        Spacer(modifier = Modifier.height(4.dp))

        Row(
            modifier = Modifier.fillMaxWidth(),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Text(
                text = "Shoe",
                color = Color.White.copy(alpha = 0.5f),
                fontSize = 11.sp
            )
            Spacer(modifier = Modifier.width(8.dp))
            LinearProgressIndicator(
                progress = { 1f - shoePenetration },
                modifier = Modifier
                    .weight(1f)
                    .height(6.dp),
                color = if (shoePenetration > 0.7f) Color(0xFFFF9800) else GoldAccent,
                trackColor = Color.White.copy(alpha = 0.1f),
                drawStopIndicator = {}
            )
        }

        Spacer(modifier = Modifier.height(6.dp))

        Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
            OutlinedButton(onClick = onEnd) {
                Text("End", color = Color.White)
            }
            OutlinedButton(onClick = onShowChart) {
                Text("Chart", color = Color.White)
            }
        }
    }
}

File: ios/Blackjack/Blackjack/Views/Components/GameInfoBar.swift
import SwiftUI

struct GameInfoBar: View {
    let chips: Int
    let shoePenetration: Float
    let handsPlayed: Int
    let handsWon: Int
    let coachEnabled: Bool
    let onToggleCoach: () -> Void
    let deviationsEnabled: Bool
    let onToggleDeviations: () -> Void
    let showCount: Bool
    let runningCount: Int
    let trueCount: Float
    let onToggleCount: () -> Void
    let onShowChart: () -> Void
    let onEnd: () -> Void
    var coachFeedback: String = ""
    var coachCorrect: Int = 0
    var coachTotal: Int = 0

    var body: some View {
        VStack(spacing: 4) {
            // Top row: chips, win ratio, toggles
            HStack {
                HStack(spacing: 4) {
                    Text("Chips:")
                        .font(.system(size: 14))
                        .foregroundColor(.white.opacity(0.7))
                    Text("$\(chips)")
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(.goldAccent)
                }

                Spacer()

                Text("W: \(handsWon) / \(handsPlayed)")
                    .font(.system(size: 13))
                    .foregroundColor(.white.opacity(0.7))

                Spacer()

                HStack(spacing: 6) {
                    TogglePill(
                        label: "Coach",
                        isOn: coachEnabled,
                        activeColor: Color(hex: 0x4CAF50),
                        action: onToggleCoach
                    )

                    if coachEnabled {
                        TogglePill(
                            label: "Dev",
                            isOn: deviationsEnabled,
                            activeColor: Color(hex: 0xFF9800),
                            action: onToggleDeviations
                        )
                    }

                    let trueCountStr = trueCount >= 0 ? String(format: "+%.1f", trueCount) : String(format: "%.1f", trueCount)
                    let rcStr = runningCount >= 0 ? "+\(runningCount)" : "\(runningCount)"
                    TogglePill(
                        label: showCount ? "RC:\(rcStr) TC:\(trueCountStr)" : "Count",
                        isOn: showCount,
                        activeColor: Color(hex: 0x42A5F5),
                        action: onToggleCount
                    )
                }
            }

            // Shoe progress
            HStack(spacing: 8) {
                Text("Shoe")
                    .font(.system(size: 11))
                    .foregroundColor(.white.opacity(0.5))
                ProgressView(value: Double(1 - shoePenetration))
                    .tint(shoePenetration > 0.7 ? Color(hex: 0xFF9800) : .goldAccent)
                    .frame(height: 6)
            }

            // Bottom row: buttons + coach feedback
            HStack(spacing: 8) {
                Button("End") { onEnd() }
                    .font(.system(size: 14))
                    .foregroundColor(.white)
                    .padding(.horizontal, 12)
                    .padding(.vertical, 6)
                    .overlay(
                        RoundedRectangle(cornerRadius: 8)
                            .stroke(Color.white.opacity(0.5), lineWidth: 1)
                    )

                Button("Chart") { onShowChart() }
                    .font(.system(size: 14))
                    .foregroundColor(.white)
                    .padding(.horizontal, 12)
                    .padding(.vertical, 6)
                    .overlay(
                        RoundedRectangle(cornerRadius: 8)
                            .stroke(Color.white.opacity(0.5), lineWidth: 1)
                    )

                if coachEnabled && !coachFeedback.isEmpty {
                    let isCorrect = coachFeedback.hasPrefix("Correct")
                    let suffix = coachTotal > 0 ? " (\(coachCorrect)/\(coachTotal))" : ""
                    Text(coachFeedback + suffix)
                        .font(.system(size: 12, weight: .bold))
                        .foregroundColor(isCorrect ? Color(hex: 0x4CAF50) : Color(hex: 0xFF9800))
                        .lineLimit(2)
                }

                Spacer()
            }
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 8)
        .background(Color.feltGreenDark)
    }
}

private struct TogglePill: View {
    let label: String
    let isOn: Bool
    let activeColor: Color
    let action: () -> Void

    var body: some View {
        Text(label)
            .font(.system(size: 12, weight: isOn ? .bold : .regular))
            .foregroundColor(isOn ? activeColor : .white.opacity(0.4))
            .padding(.horizontal, 8)
            .padding(.vertical, 3)
            .background(
                isOn
                    ? activeColor.opacity(0.2)
                    : Color.clear
            )
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .stroke(isOn ? Color.clear : Color.white.opacity(0.3), lineWidth: 1)
            )
            .clipShape(RoundedRectangle(cornerRadius: 12))
            .onTapGesture { action() }
    }
}
